packages/@sanity/components/src/selects/StyleSelect.js

```diff
@@ -111,63 +77,20 @@ class StyleSelect extends React.Component {
     }
   }
 
-
-  handleKeyDown = event => {
-    const {items} = this.props
-    const {arrowNavigationPosition} = this.state
-    if (items) {
-      if (event.key == 'ArrowUp' && arrowNavigationPosition > 0) {
-        this.setState({
-          arrowNavigationPosition: arrowNavigationPosition - 1,
-          inputValue: items[arrowNavigationPosition - 1].title,
-          showList: true
-        })
-        return false
-      }
-
-      if (event.key == 'ArrowDown' && arrowNavigationPosition < items.length - 1) {
-        this.setState({
-          arrowNavigationPosition: arrowNavigationPosition + 1,
-          inputValue: items[arrowNavigationPosition + 1].title,
-          showList: true
-        })
-        return false
-      }
-    }
-    return true
-  }
-
-  handleKeyUp = event => {
-    const {items} = this.props
-    const {arrowNavigationPosition} = this.state
-    if (event.key == 'Enter' && arrowNavigationPosition) {
-      this.handleSelect(items[arrowNavigationPosition])
-      this.setState({
-        hasFocus: false
-      })
-      return false
-    }
-    return true
-  }
-
-  componentWillMount() {
-    this._inputId = uniqueId('CustomSelect')
+  handleRootKeyPress = () => {
+    this.handleOpenList()
   }
 
   render() {
-    const {error, value, description, items, className, transparent, placeholder} = this.props
-    const {hasFocus, showList} = this.state
-
+    const {value, transparent, items, className, placeholder} = this.props
+    const {showList} = this.state
 
     return (
       <div
-        className={`
-          ${styles.root}
-          ${hasFocus ? styles.focused : ''}
-          ${error ? styles.error : ''}
-          ${transparent ? styles.transparent : ''}
-          ${className || ''}
-        `}
+        className={`${styles.root} ${className || ''} ${transparent ? styles.transparent : ''}`}
+        onBlur={this.handleCloseList}
```

Comment:
Does this make the click-outside listener obsolete?

Reply:
I tried to test without the click-outisde. Does not work. Lets keep it